src/logging/Manager.cc

```diff
@@ -1730,6 +1731,88 @@ WriterFrontend* Manager::CreateWriterForFilter(Filter* filter, const std::string
+        reporter->Error("Failed to find stream for !");
+        return false;
+    }
+
+    Filter* filter = nullptr;
+
+    // Find a filter with a matching filter name.
+    for ( const auto& f : stream->filters ) {
+        if ( f->name == header.filter_name ) {
+            filter = f;
+            break;
+        }
+    }
+    if ( ! filter ) {
+        reporter->Error("Remote write failed: Did not find filter '%s' for stream '%s'", header.filter_name.c_str(),
+                        header.stream_name.c_str());
+        return false;
+    }
+
+    // Basic validation of incoming log record with local filter configuration.
+    if ( static_cast<int>(header.fields.size()) != filter->num_fields ) {
+        reporter->Error("Remote write failed: Local filter '%s' of stream '%s' has '%d' fields, got %zu",
+                        filter->name.c_str(), obj_desc_short(header.stream_id.get()).c_str(), filter->num_fields,
+                        header.fields.size());
+        return false;
+    }
+
+    // Schema checking of filter and received header.
+    for ( int i = 0; i < filter->num_fields; ++i ) {
+        const auto ft = filter->fields[i]->type;
+        const auto ht = header.fields[i].type;
+        if ( ft != ht ) {
+            reporter->Error("Remote write failed: fields[%d] got type %s, want %s for stream '%s' and filter '%s'", i,
+                            zeek::type_name(ht), zeek::type_name(ft), header.stream_name.c_str(), filter->name.c_str());
+            return false;
+        }
+    }
+
+    if ( header.writer_id.get() != filter->writer ) {
+        reporter->Error("Remote write failed: Local filter '%s' of stream '%s' has writer '%s', got '%s'",
+                        filter->name.c_str(), header.stream_name.c_str(), obj_desc_short(filter->writer).c_str(),
+                        header.writer_name.c_str());
+        return false;
+    }
+
+    // Lookup if there's a (writer, path) pair for this stream. Either use
+    // it or create a new one if not.
+    //
+    // TODO: Might make sense to switch this to (stream, filter, path)
+    // down the road.
+    Stream::WriterPathPair wpp(filter->writer->AsEnum(), header.path);
+
+    Stream::WriterMap::const_iterator w = stream->writers.find(wpp);
+    if ( w == stream->writers.end() ) {
+        DBG_LOG(DBG_LOGGING, "creating writer %s for filter %s of stream %s\n", header.writer_name.c_str(),
+                filter->name.c_str(), header.stream_name.c_str());
+
+        if ( ! CreateWriterForFilter(filter, header.path, WriterOrigin::REMOTE) ) {
+            reporter->Error("Failed to create writer %s for filter '%s' of stream '%s'", header.writer_name.c_str(),
+                            filter->name.c_str(), header.stream_name.c_str());
+            return false;
+        }
+    }
+
+    w = stream->writers.find(wpp);
+    assert(w != stream->writers.end());
+
+    // Write each record individually. This results in the
+    // frontends buffering to be in effect.
```

Comment:
I don't fully follow this comment — you mean it builds up buffering in the frontend because we don't flush explicitly? (Is it "frontend's buffering"?)

Reply:
It's "frontend's buffering", yes. And yes, if 8 nodes send log batches of a small size at about the same time, the logger would still just flush its own buffer once, rather than 8 times.